pkg/controllers/leaderworkerset_controller.go

```diff
@@ -363,7 +368,7 @@ func (r *LeaderWorkerSetReconciler) updateConditions(ctx context.Context, lws *l
 			var leaderPod corev1.Pod
 			if err := r.Get(ctx, client.ObjectKey{Namespace: lws.Namespace, Name: sts.Name}, &leaderPod); err != nil {
 				log.Error(err, "Fetching leader pod")
-				return false, err
+				return false, client.IgnoreNotFound(err)
```

Comment:
if the leader pod doesn't exist, the worker statefulset will be deleted automatically as well. So if the worker statefulset exist, and its leader pod is not found, we definitely needs to throw errors

Reply:
@liurupeng Oh, that's embarrassing. I apologize for not considering that when looking at the code. I thought we were supposed to ignore the errors if they couldn't be found.